scantool/diag_l0_elm.c

```diff
@@ -824,6 +897,9 @@ elm_send(struct diag_l0_device *dl0d,
 	if (len <= 0)
 		return diag_iseterr(DIAG_ERR_BADLEN);
 
+	if ((dev->protocol & DIAG_L1_ISO9141) && len <= 3)
```

Comment:
Interesting; I'm not opposed to this but I wonder if you had problems related to this ?